File: Arrays/Leader_in_array.java
public class Leader_in_array {
	public static void Leader( int arr[] , int n)
	{
		int curr_leader = arr[n - 1];
		System.out.print(curr_leader+" ");
		for(int i = n -2 ; i >= 0 ; i--)
		{
			if(curr_leader < arr[i])
			{
				curr_leader = arr[i];
				System.out.print(curr_leader+" ");
			}
		}
	}
    public static void main(String[] args) {
    	int arr[] = {7, 12, 44 ,22,1,65,1, 9};
    	int n = arr.length;
    	Leader(arr , n);
    }
}

File: Arrays/Leader_in_array.cpp
// Find the leader in ana array
// Input = 16, 17, 4, 3, 5, 2
// Output = 17, 5, 2

#include<iostream>
using namespace std;
void leader(int arr[] , int n)
{
	int curr_ldr = arr[n - 1];
	cout<<curr_ldr<<" ";
	for(int i = n - 2; i >= 0 ; i --)
	{
		if(curr_ldr < arr[i])
		{
			curr_ldr = arr[i];
			cout<<curr_ldr<<" ";
		}
	}
}

int main()
{
	int arr[] = {7, 10, 4, 3, 5,12,55,3,23};
	int n = sizeof(arr) / sizeof(arr[0]);
	leader(arr , n);
}